src/code/irqmgr.c

```diff
@@ -127,54 +179,64 @@ void IrqMgr_CheckStack() {
 }
 
 void IrqMgr_HandlePRENMI450(IrqMgr* this) {
-    u64 temp = STATUS_NMI; // required to match
-    gIrqMgrResetStatus = temp;
-    this->resetStatus = STATUS_NMI;
+    u64 nmi = IRQ_RESET_NMI; // required to match
+
+    gIrqMgrResetStatus = nmi;
+    this->resetStatus = IRQ_RESET_NMI;
 
-    osSetTimer(&this->timer, OS_USEC_TO_CYCLES(30000), 0ull, &this->queue, (OSMesg)PRENMI480_MSG);
-    IrqMgr_SendMesgForClient(this, (OSMesg) & this->nmiMsg);
+    // Schedule a PRENMI480 message to be handled in 30ms
+    osSetTimer(&this->timer, OS_MSEC_TO_CYCLES(30), 0, &this->queue, (OSMesg)PRENMI480_MSG);
+    // Send the NMI event to clients
+    IrqMgr_SendMesgToClients(this, (OSMesg) & this->nmiMsg);
 }
 
 void IrqMgr_HandlePRENMI480(IrqMgr* this) {
     u32 ret;
 
-    osSetTimer(&this->timer, OS_USEC_TO_CYCLES(20000), 0ull, &this->queue, (OSMesg)PRENMI500_MSG);
+    // Schedule a PRENMI500 message to be handled in 20ms
+    osSetTimer(&this->timer, OS_MSEC_TO_CYCLES(20), 0, &this->queue, (OSMesg)PRENMI500_MSG);
+
     ret = osAfterPreNMI();
-    if (ret) {
+    if (ret != 0) {
         // "osAfterPreNMI returned %d !?"
         osSyncPrintf("osAfterPreNMIが %d を返しました！？\n", ret);
-        osSetTimer(&this->timer, OS_USEC_TO_CYCLES(1000), 0ull, &this->queue, (OSMesg)PRENMI480_MSG);
+        // osAfterPreNMI failed, try again in 1ms
+        osSetTimer(&this->timer, OS_MSEC_TO_CYCLES(1), 0, &this->queue, (OSMesg)PRENMI480_MSG);
     }
 }
 
 void IrqMgr_HandlePRENMI500(IrqMgr* this) {
-    IrqMgr_CheckStack();
+    IrqMgr_CheckStacks();
 }
 
+/**
+ * Ran on each VI retrace, measures the time elapsed between the first and second VI retrace
+ * and dispatches VI retrace messages to each registered Irq Client
+ */
 void IrqMgr_HandleRetrace(IrqMgr* this) {
-    if (gIrqMgrRetraceTime == 0ull) {
+    if (gIrqMgrRetraceTime == 0) {
         if (this->retraceTime == 0) {
             this->retraceTime = osGetTime();
         } else {
             gIrqMgrRetraceTime = osGetTime() - this->retraceTime;
         }
     }
     sIrqMgrRetraceCount++;
-    IrqMgr_SendMesgForClient(this, (OSMesg) & this->retraceMsg);
+    IrqMgr_SendMesgToClients(this, (OSMesg) & this->retraceMsg);
 }
 
-void IrqMgr_ThreadEntry(void* arg0) {
-    OSMesg msg;
-    IrqMgr* this = (IrqMgr*)arg0;
+void IrqMgr_ThreadEntry(void* arg) {
+    u32 msg = 0;
```

Comment:
I suppose 0 has no special meaning apart from "no message"?

Reply:
That's correct, by the time `msg` is used it should be replaced by one of the internal message values